File: Rich_Game/src/com/tw/commands/Roll.java
package com.tw.commands;

import com.tw.Dice;
import com.tw.player.Player;

/**
 * Created by pzzheng on 11/15/16.
 */
public class Roll implements Command {

    private Dice dice;

    public Roll(Dice dice) {
        this.dice = dice;
    }

    @Override
    public Player.Status execute(Player player) {
        return player.getMap().move(player.getCurrentPlace(), dice.next()).comeHere(player);
    }
}

File: ruby/rich_game_ruby/src/commands/roll.rb
require_relative 'command'
class Roll < Command

  def initialize(dice)
    @dice = dice
  end

  def execute(player)
    targetPlace = player.game.map.move(player.currentPlace, @dice.next)
    return targetPlace.tool.passOn(targetPlace, player) if targetPlace.tool != nil
    return targetPlace.comeHere(player)
  end


end